### DISCUSSION TITLE
GDB Debugging features

### DISCUSSION THREAD

**Discussion by prb28** [dirkwhoffmann/vAmiga#667] [Ideas]
Hi Dirk,
Can you give me some hints, I would like to add two things:

- How to store and retrieve the stack frames (backtraces)
- How to step the copper process and add breakpoints

I have a basic stepping fork here : https://github.com/prb28/vAmiga
These two feature apart, the rest seems to work.

Thanks.

**Comment by dirkwhoffmann** [comment]
Both features are not implemented, yet.

For the stack frame question, I need to dig deeper into the theory. Ad hoc, I don't know what exactly has to be done. Copper breakpoints are planned for v2.1. I know what to do, but didn't have time to implement it yet.

**Reply by prb28** [reply]
For the stack, I don't know what is a standard stack trace for a debugger and how the C stack trace of gdb is managed.
In my case, I need for each jump to subroutine to save the registers so I can display them and the address of the jump (to sub routine) instruction.

**Reply by dirkwhoffmann** [reply]
This means I need to record the new PC and the current register contents whenever a JSR or BSR instruction is executed. What about interrupts and exceptions? Are they supposed to be recorded, too?

**Reply by prb28** [reply]
I don't think they need to be recorded...
The exception should be caught and the address returned to the debugger is the position where the exception was thrown.
The stack trace should show all the branch path to this point.
But if the exception is not caught, it is just forgotten. But I can be wrong.
